test/e2e/custom-networking/custom_networking_suite_test.go

```diff
@@ -0,0 +1,201 @@
+
+		subnetCidr, err := cidr.Subnet(cidrRange, 8, 5*i)
+		Expect(err).ToNot(HaveOccurred())
+
+		createSubnetOutput, err := f.CloudServices.EC2().
+			CreateSubnet(subnetCidr.String(), f.Options.AWSVPCID, az)
+		Expect(err).ToNot(HaveOccurred())
+
+		subnetID := *createSubnetOutput.Subnet.SubnetId
+
+		By("associating the route table with the newly created subnet")
+		err = f.CloudServices.EC2().
+			AssociateRouteTableToSubnet(clusterVPCConfig.PublicRouteTableID, subnetID)
+		Expect(err).ToNot(HaveOccurred())
+
+		eniConfig, err := manifest.NewENIConfigBuilder().
+			Name(az).
+			SubnetID(subnetID).
+			SecurityGroup([]string{customNetworkingSGID}).
+			Build()
+		Expect(err).ToNot(HaveOccurred())
+
+		By("creating the ENIConfig with az name")
+		err = f.K8sResourceManagers.CustomResourceManager().CreateResource(eniConfig)
+		Expect(err).ToNot(HaveOccurred())
+
+		// For deleting later
+		customNetworkingSubnetIDList = append(customNetworkingSubnetIDList, subnetID)
+		eniConfigList = append(eniConfigList, eniConfig)
+	}
+
+	By("enabling custom networking on aws-node DaemonSet")
+	k8sUtils.AddEnvVarToDaemonSetAndWaitTillUpdated(f, "aws-node",
+		"kube-system", "aws-node", map[string]string{
+			"AWS_VPC_K8S_CNI_CUSTOM_NETWORK_CFG": "true",
+			"ENI_CONFIG_LABEL_DEF":               "failure-domain.beta.kubernetes.io/zone",
+			"WARM_ENI_TARGET":                    "0",
+		})
+
+	nodeGroupProperties = awsUtils.NodeGroupProperties{
+		NgLabelKey:                "node-type",
+		NgLabelVal:                "custom-networking-node",
+		AsgSize:                   2,
+		NodeGroupName:             "custom-networking-node",
+		IsCustomNetworkingEnabled: true,
+		Subnet:                    clusterVPCConfig.PublicSubnetList,
+		InstanceType:              "c5.xlarge",
+		KeyPairName:               keyPairName,
+	}
+
+	By("creating a new self managed node group")
+	err = awsUtils.CreateAndWaitTillSelfManagedNGReady(f, nodeGroupProperties)
+	Expect(err).ToNot(HaveOccurred())
+})
+
+var _ = AfterSuite(func() {
+	By("deleting test namespace")
+	f.K8sResourceManagers.NamespaceManager().
+		DeleteAndWaitTillNamespaceDeleted(utils.DefaultTestNamespace)
+
+	By("waiting for some time to allow CNI to delete ENI for IP being cooled down")
+	time.Sleep(time.Second * 60)
+
+	By("deleting the self managed node group")
+	err = awsUtils.DeleteAndWaitTillSelfManagedNGStackDeleted(f, nodeGroupProperties)
+	Expect(err).ToNot(HaveOccurred())
+
+	By("deleting the key pair")
+	f.CloudServices.EC2().DeleteKey(keyPairName)
+
+	err = f.CloudServices.EC2().DeleteSecurityGroup(customNetworkingSGID)
+	Expect(err).ToNot(HaveOccurred())
+
+	for _, subnet := range customNetworkingSubnetIDList {
+		By(fmt.Sprintf("deleting the subnet %s", subnet))
+		err = f.CloudServices.EC2().DeleteSubnet(subnet)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	By("disassociating the CIDR range to the VPC")
+	err = f.CloudServices.EC2().DisAssociateVPCCIDRBlock(cidrBlockAssociationID)
```

Comment:
Won't we delete the VPC?

Reply:
For all the e2e test, I was thinking of having a central entry point which creates the cluster and runs all e2e test and after execution of the e2e suites the entrypoint would delete all cluster resources.

Reply:
sure that would work too :)